### ISSUE TITLE
Rack crash CV-MAP / PAM

### ISSUE THREAD

**Issue by Coirt** [stoermelder/vcvrack-packone#31]
Windows 7, Rack 1.1.3

When Map / Pam is in mapping mode and the module is deleted with `backspace` rack crashes.

To reproduce enter mapping mode on any slot and hit backspace.

Crash does not occur when the deleting through the context menu. Probably because of the focus of "mapping..." when left mouse button is pressed.
_Core CV-MAP does not crash._

stack trace:
```
[154.753 fatal src/main.cpp:38] Fatal signal 11. Stack trace:
31: ZN4rack10appDestroyEv 0x456415
30: gai_strerrorW 0x757ac0
29: _C_specific_handler 0x774872fc
28: _chkstk 0x7749bee0
27: RtlInitializeResource 0x7746ff00
26: KiUserExceptionDispatcher 0x7749b5f0
25: ZN15MapModuleChoiceILi32EE10onDeselectERKN4rack5event8DeselectE 0xc610920
24: ZN4rack5event5State14finalizeWidgetEPNS_6widget6WidgetE 0x4574f0
23: ZN4rack6widget6Widget13clearChildrenEv 0x4a7160
22: ZN4rack6widget6Widget13clearChildrenEv 0x4a7160
21: ZN4rack6widget6WidgetD1Ev 0x4a7292
20: ZN4rack2ui12ScrollWidgetD0Ev 0x76b3a0
19: ZN4rack6widget6Widget13clearChildrenEv 0x4a7160
18: ZN4rack6widget6WidgetD1Ev 0x4a7292
17: ZN16MapModuleDisplayILi32EED0Ev 0xc612d30
16: ZN4rack6widget6Widget13clearChildrenEv 0x4a7160
15: ZN4rack6widget6WidgetD1Ev 0x4a7292
14: ZN11CVMapWidgetD0Ev 0xc607dd0
13: ZN4rack3app12ModuleWidget10onHoverKeyERKNS_5event8HoverKeyE 0x494582
12: ZN4rack6widget6Widget10onHoverKeyERKNS_5event8HoverKeyE 0x78fd00
11: ZN4rack3app10RackWidget10onHoverKeyERKNS_5event8HoverKeyE 0x48fdda
10: ZN4rack6widget10ZoomWidget10onHoverKeyERKNS_5event8HoverKeyE 0x78e950
9: ZN4rack6widget6Widget10onHoverKeyERKNS_5event8HoverKeyE 0x78fd00
8: ZN4rack3app16RackScrollWidget10onHoverKeyERKNS_5event8HoverKeyE 0x48bbc8
7: ZN4rack3app5Scene10onHoverKeyERKNS_5event8HoverKeyE 0x491a78
6: ZN4rack5event5State11handleHoverENS_4math3VecES3_ 0x457b60
5: ZN4rack6Window3runEv 0x4594ca
4: main 0x7a0000
3: main 0x7a0000
2: main 0x7a0000
1: BaseThreadInitThunk 0x772259c0
0: RtlUserThreadStart 0x77483840
```

**Comment by Coirt** [comment]
Here is the `make debug` seg fault:
```
Thread 1 received signal SIGSEGV, Segmentation fault.
MapModuleChoice<32, CVPam>::onDeselect (this=<optimized out>, e=...)
    at src/MapModule.hpp:291
291     src/MapModule.hpp: No such file or directory.
(gdb) bt
#0  MapModuleChoice<32, CVPam>::onDeselect (this=<optimized out>, e=...)
    at src/MapModule.hpp:291
#1  0x00000000004537c8 in rack::event::State::setSelected (w=0x0,
    this=0x40cd7e0) at src/event.cpp:110
#2  rack::event::State::finalizeWidget (this=0x40cd7e0, w=w@entry=0xd6d9980)
    at src/event.cpp:111
#3  0x00000000004a3d56 in rack::widget::Widget::clearChildren (
    this=this@entry=0xd6d9610) at src/widget/Widget.cpp:129
#4  0x00000000004a3d7c in rack::widget::Widget::~Widget (this=0xd6d9610,
    __in_chrg=<optimized out>) at src/widget/Widget.cpp:13
#5  rack::widget::Widget::~Widget (this=0xd6d9610, __in_chrg=<optimized out>)
    at src/widget/Widget.cpp:14
#6  rack::widget::Widget::clearChildren (this=this@entry=0xd6d9580)
    at src/widget/Widget.cpp:131
#7  0x00000000004a3e39 in rack::widget::Widget::~Widget (this=0xd6d9580,
    __in_chrg=<optimized out>) at src/widget/Widget.cpp:13
#8  0x0000000000576f87 in rack::widget::OpaqueWidget::~OpaqueWidget (
    this=0xd6d9580, __in_chrg=<optimized out>)
    at include/widget/OpaqueWidget.hpp:12
#9  rack::ui::ScrollWidget::~ScrollWidget (this=0xd6d9580,
    __in_chrg=<optimized out>) at include/ui/ScrollWidget.hpp:12
#10 rack::ui::ScrollWidget::~ScrollWidget (this=0xd6d9580,
    __in_chrg=<optimized out>) at include/ui/ScrollWidget.hpp:12
#11 0x00000000004a3ded in rack::widget::Widget::clearChildren (
    this=this@entry=0xd6d9300)
    at C:/msys64/mingw64/include/c++/8.2.1/bits/stl_list.h:153
#12 0x00000000004a3e39 in rack::widget::Widget::~Widget (this=0xd6d9300,
    __in_chrg=<optimized out>) at src/widget/Widget.cpp:13
#13 0x0000000008a03bd7 in rack::widget::OpaqueWidget::~OpaqueWidget (
    this=0xd6d9300, __in_chrg=<optimized out>)
    at ../../include/widget/OpaqueWidget.hpp:12
#14 rack::app::LedDisplay::~LedDisplay (this=0xd6d9300,
    __in_chrg=<optimized out>) at ../../include/app/LedDisplay.hpp:12
#15 MapModuleDisplay<32, CVPam, MapModuleChoice<32, CVPam> >::~MapModuleDisplay (this=0xd6d9300, __in_chrg=<optimized out>) at src/MapModule.hpp:402
#16 MapModuleDisplay<32, CVPam, MapModuleChoice<32, CVPam> >::~MapModuleDisplay (this=0xd6d9300, __in_chrg=<optimized out>) at src/MapModule.hpp:402
#17 0x00000000004a3ded in rack::widget::Widget::clearChildren (
    this=this@entry=0xd6d4ef0)
    at C:/msys64/mingw64/include/c++/8.2.1/bits/stl_list.h:153
#18 0x00000000004a3e39 in rack::widget::Widget::~Widget (this=0xd6d4ef0,
    __in_chrg=<optimized out>) at src/widget/Widget.cpp:13
#19 0x00000000089f7547 in CVPamWidget::~CVPamWidget (this=0xd6d4ef0,
    __in_chrg=<optimized out>) at src/CVPam.cpp:99
#20 CVPamWidget::~CVPamWidget (this=0xd6d4ef0, __in_chrg=<optimized out>)
    at src/CVPam.cpp:99
#21 0x0000000000491520 in rack::app::ModuleWidget::onHoverKey (
    this=0xd6d4ef0, e=...) at src/app/ModuleWidget.cpp:377
#22 0x000000000059a52b in rack::widget::Widget::recursePositionEvent<void (rack::widget::Widget::*)(rack::event::HoverKey const&), rack::event::HoverKey> (
    e=..., f=&virtual table offset 88, this=0x40d5500) at include/math.hpp:191
#23 rack::widget::Widget::onHoverKey (this=0x40d5500, e=...)
    at include/widget/Widget.hpp:145
#24 0x000000000048cde0 in rack::widget::Widget::recursePositionEvent<void (rack::widget::Widget::*)(rack::event::HoverKey const&), rack::event::HoverKey> (
    e=..., f=&virtual table offset 88, this=0x40d52f0) at include/math.hpp:191
#25 rack::widget::Widget::onHoverKey (e=..., this=0x40d52f0)
    at include/widget/Widget.hpp:145
#26 rack::widget::OpaqueWidget::onHoverKey (e=..., this=0x40d52f0)
    at include/widget/OpaqueWidget.hpp:30
#27 rack::app::RackWidget::onHoverKey (this=0x40d52f0, e=...)
    at src/app/RackWidget.cpp:127
#28 0x00000000005991c0 in rack::widget::Widget::recursePositionEvent<void (rack::widget::Widget::*)(rack::event::HoverKey const&), rack::event::HoverKey> (
    e=<synthetic pointer>..., f=&virtual table offset 88, this=0x40d5230)
    at include/math.hpp:191
#29 rack::widget::Widget::onHoverKey (e=<synthetic pointer>...,
    this=0x40d5230) at include/widget/Widget.hpp:145
#30 rack::widget::ZoomWidget::onHoverKey (this=0x40d5230, e=...)
    at include/widget/ZoomWidget.hpp:31
#31 0x000000000059a52b in rack::widget::Widget::recursePositionEvent<void (rack::widget::Widget::*)(rack::event::HoverKey const&), rack::event::HoverKey> (
    e=..., f=&virtual table offset 88, this=0x40d4fd0) at include/math.hpp:191
#32 rack::widget::Widget::onHoverKey (this=0x40d4fd0, e=...)
    at include/widget/Widget.hpp:145
#33 0x0000000000488d56 in rack::widget::Widget::recursePositionEvent<void (rack::widget::Widget::*)(rack::event::HoverKey const&), rack::event::HoverKey> (
    e=..., f=&virtual table offset 88, this=0x40d4f20) at include/math.hpp:191
#34 rack::widget::Widget::onHoverKey (e=..., this=0x40d4f20)
    at include/widget/Widget.hpp:145
#35 rack::widget::OpaqueWidget::onHoverKey (e=..., this=0x40d4f20)
    at include/widget/OpaqueWidget.hpp:30
#36 rack::app::RackScrollWidget::onHoverKey (this=0x40d4f20, e=...)
    at src/app/RackScrollWidget.cpp:85
#37 0x000000000048ea14 in rack::widget::Widget::recursePositionEvent<void (rack::widget::Widget::*)(rack::event::HoverKey const&), rack::event::HoverKey> (
    e=..., f=&virtual table offset 88, this=0x40cd4b0) at include/math.hpp:191
#38 rack::widget::Widget::onHoverKey (e=..., this=0x40cd4b0)
    at include/widget/Widget.hpp:145
#39 rack::widget::OpaqueWidget::onHoverKey (e=..., this=0x40cd4b0)
    at include/widget/OpaqueWidget.hpp:30
#40 rack::app::Scene::onHoverKey (this=0x40cd4b0, e=...)
    at src/app/Scene.cpp:59
#41 0x000000000045407a in rack::event::State::handleKey (action=3, mods=0,
    scancode=<optimized out>, key=259, pos=..., this=0x40cd7e0)
    at C:/msys64/mingw64/include/c++/8.2.1/bits/stl_tree.h:364
#42 rack::event::State::handleHover (this=0x40cd7e0, pos=...,
    mouseDelta=mouseDelta@entry=...) at src/event.cpp:178
#43 0x000000000045578d in rack::cursorPosCallback (ypos=<optimized out>,
    xpos=<optimized out>, win=0x40cee60) at src/window.cpp:144
#44 rack::Window::run (this=0x40cea90) at src/window.cpp:334
#45 0x00000000005ab7c9 in main (argc=<optimized out>, argv=<optimized out>)
    at src/main.cpp:186
(gdb) make: *** [Makefile:65: debug] Interrupt

```

**Comment by stoermelder** [comment]
Thanks, fixed in 7c5abbaaa2b299ce049ff427711e93a07a6b4c81.
Some strange template-thing.
